frame/staking/src/pallet/mod.rs

```diff
@@ -559,6 +651,184 @@ pub mod pallet {
+		}
+
+		/// Get exposure info for a validator at a given era and page.
+		///
+		/// This builds a paged exposure from `ExposureOverview` and `ExposurePage` of the
+		/// validator. For older non-paged exposure, it returns the clipped exposure directly.
+		pub(crate) fn get_validator_exposure(
+			era: EraIndex,
+			validator: &T::AccountId,
+			page: PageIndex,
+		) -> Option<ExposureExt<T::AccountId, BalanceOf<T>>> {
+			return match <ErasStakersPaged<T>>::get(era, (validator, page)) {
+				// return clipped exposure if page zero and paged exposure does not exist
+				None if page == 0 =>
+					Some(ExposureExt::from_clipped(<ErasStakersClipped<T>>::get(era, validator))),
+
+				// return paged exposure if it exists
+				Some(exposure_page) => {
+					let overview = <ErasStakersOverview<T>>::get(&era, validator);
+					// own stake is included only once in the first page.
+					let own = if page == 0 { overview.own } else { Zero::zero() };
+
+					Some(ExposureExt {
+						exposure_overview: ExposureOverview { own, ..overview },
+						exposure_page,
+					})
+				},
+				_ => None,
+			}
+		}
+
+		/// Returns the number of pages of exposure a validator has for the given era.
+		///
+		/// This will always return at minimum one count of exposure to be backward compatible to
+		/// non-paged reward payouts.
+		pub(crate) fn get_page_count(era: EraIndex, validator: &T::AccountId) -> PageIndex {
+			<ErasStakersOverview<T>>::get(&era, validator).page_count.max(1)
+		}
+
+		/// Returns the next page that can be claimed or `None` if nothing to claim.
+		pub(crate) fn get_next_claimable_page(
+			era: EraIndex,
+			validator: &T::AccountId,
+			ledger: &StakingLedger<T>,
+		) -> Option<PageIndex> {
+			if Self::is_non_paged_exposure(era, validator) {
+				return match ledger.legacy_claimed_rewards.binary_search(&era) {
+					// already claimed
+					Ok(_) => None,
+					// Non-paged exposure is considered as a single page
+					Err(_) => Some(0),
+				}
+			}
+
+			// Find next claimable page of paged exposure.
+			let page_count = Self::get_page_count(era, validator);
+			let claimed_pages = ClaimedRewards::<T>::get(era, validator);
+			let claimed_page_count = claimed_pages.len() as PageIndex;
+
+			// find the first page that is not claimed.
+			for page in 0..claimed_page_count as PageIndex {
+				debug_assert!(page <= claimed_pages[page as usize]);
+				if page < claimed_pages[page as usize] {
+					return Some(page)
+				}
+			}
+			// all pages are claimed
+			return if claimed_page_count < page_count { Some(claimed_page_count) } else { None }
+		}
+
+		/// Checks if exposure is paged or not.
+		fn is_non_paged_exposure(era: EraIndex, validator: &T::AccountId) -> bool {
+			<ErasStakersClipped<T>>::contains_key(&era, validator)
+		}
+
+		/// Returns the maximum number of pages of exposure we can store.
+		fn get_max_exposure_page_count() -> PageIndex {
+			return <MaxExposurePageCount<T>>::get().unwrap_or_default().max(1)
```

Comment:
no need fo return.

Reply:
Seems like you need a newtype that can only be 1 or more? perhaps.

Reply:
Yeah that would be the cleanest